### ISSUE TITLE
several issues with source/tree and source/commits

### ISSUE THREAD

**Issue by antineutron** [SourceKettle/sourcekettle#461]
(Reported by Anko Anchev via email)

Hello,

I would like to report a particular behaviour in sourcekettle at the page "APP/source/commits" (actually both in "/tree" and "/commits"). It involves the controls for selecting the branch and switching between "Source" and "Commits" view. I believe it is just a combination of several minor bugs.

Firstly, the drop-down control for selecting the branch always has the newest branch selected. Even if you use that same control to view a different branch, when the new page loads the control still has the newest selected. In other words, it does not really show which branch you are viewing.

Second, when you click the "Commits" button, it goes to "commits", bit it does not respect the branch you had been viewing. It always shows the commits of the newest branch. The same applies to the "Source" button. Perhaps this is not really a bug, but it would be really helpful if those two buttons respected what branch you were currently viewing.

Lastly, the drop-down control for selecting the branch always leads to the "Source" page ("/tree"). If you use the URL and just type the correct branch name in the crumb there, the correct page is displayed. For example, the url "https://sourcekettle.ecs.soton.ac.uk/project/APP/source/commits/OLD_BRANCH" would show me what I wanted to see, but there are no buttons that lead to that page/content.

TL;DR
To summarize, the top-bar elements on the pages "APP/source/tree" and "APP/source/commits" do not take into account the currently viewed branch at all.
